GoRogue/MapGeneration/Region.cs

```diff
@@ -42,36 +42,30 @@ public Region(string name, Point northWest, Point northEast, Point southEast, Po
         #endregion
 
         #region Overrides
+
         /// <summary>
-        /// Returns a string detailing the region's corner locations.
+        /// Returns whether or not this region is completely overlapping with the other region
         /// </summary>
-        /// <returns/>
-        public override string ToString()
+        /// <param name="other">the region against which to compare</param>
+        /// <returns></returns>
+        public bool Matches(Region other)
```

Comment:
Matching requires the input type to be nullable: this needs to be:
```suggestion
        public bool Matches(Region? other)
```

And needs to account for the input parameter possibly being null.  This should get rid of the requisite warnings during CI as well.

Reply:
Excellent catch. Fixed in new version